packages/wallet-manager/src/WalletManagerContext.ts

```diff
@@ -0,0 +1,95 @@
+import { provide, createContext } from '@lit/context';
+import { LitElement, html } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import {
+  type WalletManagerController as TWalletManagerController,
+  WalletManagerController
+} from './WalletManagerController';
+import { ethers } from 'ethers';
+import { ApiPromise } from '@polkadot/api';
+import { SyntheticEventCreator } from './types';
+
+export const WalletManagerContext = createContext<
+  TWalletManagerController | undefined
+>('wallet-context');
+
+export const SyntheticEventCreatorContext =
+  createContext<SyntheticEventCreator>('synthetic-event');
+
+/**
+ * @name WalletManagerContextProvider
+ * @description This component is responsible for providing the WalletManagerController as a context to all its children.
+ * It also provides a synthetic event creator function that can be used to dispatch custom events.
+ * @example
+ * For you to consume the context objects, you need to wrap up your component with the wallet-manager-context on your render method.
+ * You can either pass a web3Provider, an apiPromise or a wssConnectionUrl to the component.
+ *
+ * Passing a web3Provider and an apiPromise
+ * <wallet-manager-context
+ *   .web3Provider=${web3Provider}
+ *   .apiPromise=${apiPromise}
+ * >
+ *    <your-component></your-component>
+ * </wallet-manager-context>
+ *
+ * Passing a wssConnectionUrl
+ * <wallet-manager-context
+ *  .wssConnectionUrl=${wssConnectionUrl}
+ * >
+ *   <your-component></your-component>
+ * </wallet-manager-context>
+ */
+@customElement('wallet-manager-context')
+export class WalletManagerContextProvider extends LitElement {
+  @provide({ context: WalletManagerContext })
+  @state()
+  walletManagerController?: WalletManagerController;
+
+  @provide({ context: SyntheticEventCreatorContext })
+  @state()
+  syntheticEventCreator: SyntheticEventCreator = (
+    eventName: string,
+    dataToPass: unknown
+  ) => {
+    const event = new CustomEvent(eventName, {
+      bubbles: true,
+      composed: true,
+      detail: dataToPass
+    });
+
+    this.dispatchEvent(event);
+  };
```

Comment:
Does this need to be exposed at the context level? Would just having a small helper function that can be imported where needed not be sufficient?

Reply:
yeah actually that makes sense, since the syntetic event is being attached globally. We don't need that property as part of the context. I'm going to move it as helper function

Reply:
moved to utills folder